src/components/userSearchBar/UserSearchBar.js

```diff
@@ -0,0 +1,53 @@
+import Card from "../card";
+import TextInput from "../form/textInput";
+import SearchIcon from "../../assets/icons/searchIcon";
+import { useEffect, useState } from "react";
+import { getUserByName } from "../../service/apiClient";
+import UserCard from "../userCard";
+
+const UserSearchBar = () => {
+  const [searchVal, setSearchVal] = useState("");
+  const [searchResults, setSearchResults] = useState([]);
+
+  const onChange = (e) => {
+    setSearchVal(e.target.value);
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      try {
+        const res = await getUserByName(searchVal);
+        setSearchResults(res);
+      } catch (error) {
+        console.error("Error fetching user:", error);
```

Comment:
I think we should display this error to the user. If something goes wrong, the user won't know why the search hasn't worked.

Reply:
That error should be handled by whoever is in charge of "display results"?or am I wrong? I don't want to step into someone else work...

Reply:
I think it should be displayed in your Card, just like the search results are